standard/expressions.md

```diff
@@ -3875,6 +3876,74 @@ A lifted ([§12.4.8](expressions.md#1248-lifted-operators)) form of the range op
 
 The range operator is non-associative ([§12.4.2](expressions.md#1242-operator-precedence-and-associativity)).
 
+## §switch-expression-new-clause Switch expression
+
+A *switch_expression* provides `switch`-like semantics in an expression context.
+
+```ANTLR
+switch_expression
+    : range_expression
+    | switch_expression 'switch' '{' (switch_expression_arms ','?)? '}'
+    ;
+switch_expression_arms
+    : switch_expression_arm
+    | switch_expression_arms ',' switch_expression_arm
+    ;
+switch_expression_arm
+    : pattern case_guard? '=>' switch_expression_arm_expression
+    ;
+switch_expression_arm_expression
+    : expression
+    ;
+```
+
+A *switch_expression* is not an *expression_statement*.
+
+There is a *switch expression conversion* (§switch-expression-conversion) from a switch expression to a type `T`
+if there is an implicit conversion from every *switch_expression_arm_expression* of each of the switch expression's *switch_expression_arm*s to `T`.
+
+If a switch expression is not subject to a *switch expression conversion*, then
+
+- The type of the *switch_expression* is the best common type [§12.6.3.15](expressions.md#126315-finding-the-best-common-type-of-a-set-of-expressions)) of the *switch_expression_arm_expression*s of the *switch_expression_arm*s if such a type exists and each *switch_expression_arm_expression* can be implicitly converted to that type.
+- It is an error if no such type exists.
+
+It is an error if some *switch_expression_arm*'s pattern cannot affect the result because some previous pattern and guard will always match.
+
+A switch expression is said to be *exhaustive* if every value of its input is handled by at least one arm of the switch expression.  The compiler shall produce a warning if a switch expression is not exhaustive.
```

Comment:
style question, do we need "said to be"?, or is this the definition?

```suggestion
A switch expression is *exhaustive* if every value of its input is handled by at least one arm of the switch expression.  The compiler shall produce a warning if a switch expression is not exhaustive.
```

Reply:
An addition to @BillWagner’s change – remove the compiler, also add blank line at end to break para. (Includes Bill’s suggestion):

```suggestion
A switch expression is *exhaustive* if every value of its input is handled by at least one arm of the switch expression.  A warning shall be issued if a switch expression is not exhaustive.

```

Reply:
Let's change "shall be" to "may be" which also gives us wiggle room around nullable types and exhaustiveness.